implementations/rust/ockam/ockam_app/src/app/tray_menu.rs

```diff
@@ -0,0 +1,118 @@
+use crate::enroll::EnrollTrayMenuSection;
+use crate::options::OptionsTrayMenuSection;
+use crate::tcp::outlet::TcpOutletTrayMenuSection;
+use crate::{enroll, options, tcp};
+
+use ockam_core::async_trait;
+
+use crate::app::AppState;
+use tauri::{
+    AppHandle, CustomMenuItem, Manager, SystemTrayEvent, SystemTrayHandle, SystemTrayMenu,
+    SystemTrayMenuItem, Wry,
+};
+use tracing::error;
+
+#[derive(Default)]
+pub struct TrayMenu {
+    pub enroll: EnrollTrayMenuSection,
+    pub tcp: TcpOutletTrayMenuSection,
+    pub options: OptionsTrayMenuSection,
+}
+
+impl TrayMenu {
+    pub fn init(&self) -> Self {
```

Comment:
> init is not necessary since there is default

👍 

> build could be private

I need to call that function from the `setup_app` function

> refresh would be enough to both create the initial version and any refreshed ones

I don't get this one. You mean building the menu out of a new menu instance instead of mutating the existing one?

Reply:
> I need to call that function from the setup_app function

Isn't `lock.init().build(state.is_enrolled())` the same as`lock.refresh(app_handle)` on start-up?

> I don't get this one. You mean building the menu out of a new menu instance instead of mutating the existing one?

Don't sweat it, I'll try to make a pass on this next week to see if I can propose anything different.

Reply:
> Isn't lock.init().build(state.is_enrolled()) the same aslock.refresh(app_handle) on start-up?

Not exactly. Note that after replacing `init` by default, we no longer need an async runtime to initialize it as it just creates an instance without any list item (it only includes the static items that do not depend on any node state).

The `refresh` function does need the async runtime to query data from the node.